src/cross_section.F90

```diff
@@ -141,99 +157,133 @@ subroutine calculate_nuclide_xs(i_nuclide, i_sab, E, i_mat, i_nuc_mat, i_log_uni
     integer, intent(in) :: i_nuc_mat ! index into nuclides array for a material
     integer, intent(in) :: i_log_union ! index into logarithmic mapping array or
                                        ! material union energy grid
+    real(8), intent(in) :: sqrtkT    ! Square root of kT, material dependent
 
     integer :: i_grid ! index on nuclide energy grid
     integer :: i_low  ! lower logarithmic mapping index
     integer :: i_high ! upper logarithmic mapping index
     real(8) :: f      ! interp factor on nuclide energy grid
+    real(8) :: sigT, sigA, sigF ! Intermediate multipole variables
     type(NuclideCE), pointer :: nuc
-    type(Material),   pointer :: mat
+    type(Material),  pointer :: mat
 
     ! Set pointer to nuclide and material
     nuc => nuclides(i_nuclide)
     mat => materials(i_mat)
 
-    ! Determine index on nuclide energy grid
-    select case (grid_method)
-    case (GRID_MAT_UNION)
+    ! If MP, don't interpolate, it's all already baked in.
+    if (nuc % mp_present .and. &
+         (E >= nuc % multipole % start_E/1.0e6_8 .and.&
+          E <= nuc % multipole % end_E/1.0e6_8)) then
 
-      i_grid = mat % nuclide_grid_index(i_nuc_mat, i_log_union)
+      ! Call multipole kernel
+      call multipole_eval(nuc % multipole, E, sqrtkT, sigT, sigA, sigF)
 
-    case (GRID_LOGARITHM)
-      ! Determine the energy grid index using a logarithmic mapping to reduce
-      ! the energy range over which a binary search needs to be performed
+      micro_xs(i_nuclide) % total = sigT
+      micro_xs(i_nuclide) % absorption = sigA
+      micro_xs(i_nuclide) % elastic = sigT - sigA
```

Comment:
Just did analysis.  I was under the impression that inelastic scattering below the RRR/URR cutoff was not supposed to happen.  Basically, since the LRF=3 form of Reich-Moore does not support inelastic channels, that anyone using LRF=3 as a curve fitting basis would be doing things incorrectly.  Whether or not it is correct is irrelevant, though, since at least Fe-57 has MT=4 below RRR/URR cutoff.

Currently in discussion on how to fix this, but right now, nuclides in which MT=4 is below RRR/URR (and similarly, all non-(n,gamma) MT=101(?) channels) should just be treated as pointwise and should not be converted into windowed multipole, as the solution is likely to require something other than windowed multipole to fix, such as Pablo's new pointwise algorithm.